### PR TITLE
Feature: Template Checker

### PR DESCRIPTION
**Pull request by i-am-offline**
First development of template checker.
Based on issue #64 
The template checker can be found as additional menu item in the config page.
It checks all pages of the currently choosen page set.
Pagechecker requires php-dom module. In case this module is not available, all checks fail with an according message
Currently the following things are checked

- basic HTML validation (errors thrown when php-dom parses the file)
- existence of images used in IMG tags
- existence of images used in widgets

### CODE REVIEW

File: lib/templatechecker/templatechecker.css [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	.severity_N, .severity_I, .severity_W, .severity_E, .severity_U {
+		background-size: 36px 36px;
+		background-repeat: no-repeat;
+		padding-left: 36px;
+		display: block;
+		font-size: 14px;
+		line-height: 36px;
+	}
+	.severity_E {
+		background-image: url('/icons/ws/message_stop.svg');
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.css]: should be ` background-image: url('../../icons/ws/message_stop.svg');`

File: lib/templatechecker/templatechecker.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		color: red;
+	}
+	.severity_W {
+		background-image: url('/icons/ws/message_attention.svg');
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.css]: see above

File: lib/templatechecker/templatechecker.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		color: yellow;
+	}
+	.severity_N {
+		background-image: url('/icons/ws/control_ok.svg');
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.css]: see above

File: lib/templatechecker/templatechecker.css [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		color: greenyellow;
+	}
+	.severity_I {
+		background-image: url('/icons/ws/message_info.svg');
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.css]: see above

File: lib/templatechecker/templatechecker.css [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+		color: greenyellow;
+	}
+	.severity_U {
+		background-image: url('/icons/ws/control_home.svg');
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.css]: see above

File: lib/templatechecker/templatechecker.js [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+			btn_run.hide();
+			btn_reset.show();
+		});
+	};
+
+	analyze = function (fileObject) {
+
+		$.ajax({
+			type: 'POST',
+			url: '/lib/templatechecker/templatechecker.php',
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.js]: should be `url: 'lib/templatechecker/templatechecker.php',`

File: lib/templatechecker/templatechecker.js [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+	};
+
+	analyzeError = function (xhr, textStatus, errorThrown) {
+		$('#r' + fileObject.attr('id').substring(1)).html(xhr.responseJSON.message);
+	};
+
+	reset = function (e) {
+		$.ajax({
+			type: "POST",
+			url: '/lib/templatechecker/templatechecker.php',
```
Comment (smaiLee) [inline lib/templatechecker/templatechecker.js]: see above

File: pages/base/templatechecker.html [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+* @copyright   2016
+* @license     GPL [http://www.gnu.de]
+* -----------------------------------------------------------------------------
+*/
+
+
+{% extends "system.html" %}
+
+{% block content %}
+<link rel="stylesheet" href="/lib/templatechecker/templatechecker.css">
```
Comment (smaiLee) [inline pages/base/templatechecker.html]: should be `<link rel="stylesheet" href="lib/templatechecker/templatechecker.css">`

File: pages/base/templatechecker.html [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+	<div style="width:250px; float:left;"><input type="button" value="Run all tests" id="run"/></div>
+	<div style="width:250px; float:left;"><input type="button" value="Reset result" id="reset"/></div>
+	<div style="width:250px; float:left;"><input type="button" value="Show warnings/errors only" id="show_errors"/></div>
+	<div style="width:250px; float:left;"><input type="button" value="Show all messages" id="show_all"/></div>
+	<div style="clear:both"></div>
+</div>
+<h2>Result:</h2>
+<div id="filelist" style="display:none;"></div>
+
+<script src="/lib/templatechecker/templatechecker.js" type="text/javascript"></script>
```
Comment (smaiLee) [inline pages/base/templatechecker.html]: should be `<script src="lib/templatechecker/templatechecker.js" type="text/javascript">`

**Comment by i-am-offline** [comment]
The Template Checker should now be ready for testing ...

**Pull request merged**